### PR TITLE
JSEP: Use global-agent in scripts to enable using network proxy

### PR DESCRIPTION
**Pull request by jzm-intel**
This PR add dependency to the global-agent package, and use it in JSEP scripts that download files from network (i.e. `js/scripts/utils.ts` and `js/web/script/pull-prebuilt-wasm-artifacts.ts`), so that user can make these script use network proxy by setting environment variable GLOBAL_AGENT_HTTPS_PROXY.

### CODE REVIEW

**Comment by jzm-intel** [comment]
@qjia7 @gyagp @fs-eire Please take a look, thanks

**Comment by gyagp** [comment]
We have two proxy issues: 1) Electron npm package 2) ONNX opsets during test. 
Can this PR fix both issues?
The little drawback of global-agent is we need to set another specific environmental variable GLOBAL_AGENT_HTTPS_PROXY, instead of conventional https_proxy, http_proxy, no_proxy and so on. I wonder if other solutions, like [Axios](https://learn.microsoft.com/en-us/microsoft-cloud/dev/dev-proxy/how-to/use-dev-proxy-with-nodejs#axios), could be better.

**Comment by jzm-intel** [comment]
Tried Axios but seems doesn't work for our proxy setting yet...

The known solution to handle the proxy with current PR is run commands with prefix `npx cross-env ELECTRON_GET_USE_PROXY=true GLOBAL_AGENT_HTTPS_PROXY=http://proxy-host:port [any npm command]`, e.g.
```
npx cross-env ELECTRON_GET_USE_PROXY=true GLOBAL_AGENT_HTTPS_PROXY=http://proxy-host:port npm install
```
This `ELECTRON_GET_USE_PROXY` env var will make electron also use the proxy ***via global-agent*** [ref](https://www.electronjs.org/docs/latest/tutorial/installation#proxies) (for Node 10 and above).

So since electron also use global-agent, I think using global-agent here in the ORT scripts could be reasonable.

**Comment by gyagp** [comment]
Thanks for trying Axios. So this PR only solves the ONNX opsets download issue? Anyway, this LGTM.

File: js/scripts/utils.ts [hunk 1]
```diff
@@ -2,9 +2,13 @@
 // Licensed under the MIT License.
 
 import { WriteStream } from 'fs';
+import {bootstrap as global_agent_bootstrap} from 'global-agent';
 import * as https from 'https';
 import { JSZipObject } from 'jszip';
 
+// Bootstrap global-agent to honor the proxy settings
```
Comment (fs-eire) [inline js/scripts/utils.ts:9]: can add more few comments referring to GLOBAL_AGENT_HTTPS_PROXY or documents

File: js/scripts/utils.ts [hunk 2]
```diff
@@ -2,9 +2,13 @@
...
+global_agent_bootstrap();
```
Comment (fs-eire) [inline js/scripts/utils.ts:10]: use camelCase
```suggestion
globalAgentBootstrap();
```

File: js/scripts/utils.ts (hunk 1 above)
Comment (jzm-intel) [reply js/scripts/utils.ts]: Done.

File: js/scripts/utils.ts (hunk 2 above)
Comment (jzm-intel) [reply js/scripts/utils.ts]: Done.

**Comment by jzm-intel** [comment]
> So this PR only solves the ONNX opsets download issue?

Yes, but this PR make the scripts use the same proxy environment variables as Electron, since Electron also use global-agent to handle the proxy if `ELECTRON_GET_USE_PROXY=true`.

**Review by fs-eire — APPROVED** [review]

**Pull request merged**